microsoft-edge/extensions-chromium/developer-guide/devtools-extension.md

```diff
@@ -0,0 +1,430 @@
+
+1. Create the `background.js` file and add the following code to it.
+
+    ```javascript
+    let id = null;
+    const connections = {};
+
+    chrome.runtime.onConnect.addListener(devToolsConnection => {
+        // Assign the listener function to a variable so we can remove it later
+        let devToolsListener = (message, sender, sendResponse) => {
+            if (message.name == "init") {
+                id = message.tabId;
+                connections[id] = devToolsConnection;
+                // Send a message back to DevTools
+                connections[id].postMessage("Connected!");
+            }
+        };
+
+        // Listen to messages sent from the DevTools page
+        devToolsConnection.onMessage.addListener(devToolsListener);
+
+        devToolsConnection.onDisconnect.addListener(() => {
+            devToolsConnection.onMessage.removeListener(devToolsListener);
+        });
+    });
+    ```
+
+    The above code snippet connects the background service worker with the DevTools page. It listens to when the DevTools page connects, saves the connection, and sends a response back to the DevTools page.
+
+    This is useful when your background service worker is collecting data or performing tasks in the background that you want to be available in your DevTools extension.
+
+To test the new `sayHello` button:
+
+1. Navigate (or go back) to the `edge://extensions/` page.
+
+1. Find the unpacked extension entry you loaded in step 1.
+
+1. Click **Reload**.
+
+1. Open a new browser tab, open DevTools (`F12`), and click **Sample panel**.
+
+1. Click the `"Say Hello to The Inspected Page!"` button in your panel. You should see an alert in the inspected page, as shown below.
+
+    ![Microsoft Edge, showing the new extension panel in DevTools on the side, and an alert dialog window](./devtools-extensions-images/devtools-extension-show-greeting.png)
+
+To learn more about content scripts, see [the Content Scripts documentation](https://developer.chrome.com/docs/extensions/mv3/content_scripts/).
+
+To test the `youClickedOn` label:
+
+1. Click **OK** to close the alert window opened in the previous step.
+
+1. Click anywhere in the inspected page. A **You clicked on position (x,y) in the inspected page** message should appear in the extension panel.
+
+    ![The sample panel in DevTools showing the clicked position message](./devtools-extensions-images/devtools-extension-show-position.png)
+
+1. Open the **Console** tool. A message similar to the previous step should also appear in the  **Console** tool.
+
+    ![The DevTools Console tool showing the clicked position message](./devtools-extensions-images/devtools-extension-show-position-in-console.png)
+
+
+<!-- ====================================================================== -->
+## Conclusion
```

Comment:
consider moving or copying to top of page a different variant version of this list, per mgmt.
Consider changing to list in one spot.